detectable/src/main/java/com/synopsys/integration/detectable/detectables/clang/linux/LinuxDistro.java

```diff
@@ -17,9 +19,29 @@ public class LinuxDistro {
 
     public Optional<String> extractLinuxDistroNameFromEtcDir() {
         File etcDir = new File("/etc");
+
+        if (isOracleLinux(etcDir)) {
+            logger.trace("Target image Linux distro is an Oracle Linux");
+            return Optional.of("oracle_linux");
+        }
+
         return extractLinuxDistroNameFromEtcDir(etcDir);
     }
 
+    public boolean isOracleLinux(File etcDirFile) {
+        for (File etcFile : etcDirFile.listFiles()) {
+            if ("oracle-release".equals(etcFile.getName())) {
```

Comment:
We shall discuss this in a separate call.

Reply:
@niravrsynopsys the exception of oracle linux is, it contains both `oracle_release` and `redhat_release` files. The `getLInuxDistroName` is called in the following loop:
```
for (File etcFile : etcDir.listFiles()) {
            if (isLinuxDistroFile(etcFile)) {
                Optional<String> distroAccordingToThisFile = getLinxDistroName(etcFile);
                if (distroAccordingToThisFile.isPresent()) {
                    return distroAccordingToThisFile;
                }
            }
        }
```
If the `redhat_release` file comes before `oracle_release` file in the `etcDir.listFiles` then it will return the wrong linux distro name. Considering the situation, I had to check oracle linux separately.